File: residue monitoring system backend/src/main/java/kstrinadka/residue_monitoring_system/model/Product.java
package kstrinadka.residue_monitoring_system.model;


import jakarta.persistence.*;
import lombok.*;

/**
 * Сущность товара
 */
@AllArgsConstructor
@Entity
@NoArgsConstructor
@Table(name = "products")
@Getter
@Setter
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    private Long id;

    private String item_number;

    private String name;

    private Long rest;

    private Long cost;

    @Column(name = "out_of_stock", nullable = false)
    private Long outOfStock;

    @Column(name = "sold_quantity", nullable = false)
    private Long soldQuantity;

    @ManyToOne
    @JoinColumn (name = "type_id", referencedColumnName = "id", nullable = false)
    private ProductType type;
}

File: frontend/src/app/products/dto/product.ts

export class ProductFront {
  id: number;
  item_number: string;
  name: string;
  rest: number;
  type_id: number;
  cost: number;
  outOfStock: number;
  sold_quantity: number;
}

export class Product {
  id: number;
  item_number: string;
  name: string;
  rest: number;
  type: Category;
  cost: number;
  outOfStock: number;
  soldQuantity: number;
}

export class Category {
  id: number;
  categoryname: string
}
